frontend/public/components/modals/column-management-modal.tsx

```diff
@@ -0,0 +1,186 @@
+    event.preventDefault();
+    setDefaultColumns(updateColumns(true, null, defaultColumns));
+    setAdditionalColumns(updateColumns(false, null, additionalColumns));
+  };
+
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle className="modal-header">Manage Columns</ModalTitle>
+      <ModalBody>
+        <div className="co-m-form-row">
+          <p>Checked categories will be displayed in the table.</p>
+        </div>
+        <div className="row co-m-form-row">
+          <Alert
+            className="co-alert"
+            isInline
+            title={`You can select up to ${MAX_VIEW_COLS} columns`}
+            variant="info"
+          />
+        </div>
+        <div className="row co-m-form-row">
+          <div className="col-sm-12">
+            <span className="col-sm-6">
+              <label className="control-label">{`Default ${columnManagementType} Columns`}</label>
+              <DataList aria-label="default column list" id="defalt-column-management" isCompact>
+                {defaultColumns.map((defaultColumn) => (
+                  <DataListRow
+                    key={defaultColumn.ID}
+                    onChange={defaultColumnChange}
+                    disableUncheckedRow={areMaxColumnsDisplayed}
+                    column={defaultColumn}
+                  />
+                ))}
+              </DataList>
+            </span>
+            <span className="col-sm-6">
+              <label className="control-label">Additional Columns</label>
+              <DataList
+                aria-label="Additional column list"
+                id="additional-column-management"
+                isCompact
+              >
+                {additionalColumns.map((additionalColumn) => (
+                  <DataListRow
+                    key={additionalColumn.ID}
+                    onChange={additionalColumnChange}
+                    disableUncheckedRow={areMaxColumnsDisplayed}
+                    column={additionalColumn}
+                  />
+                ))}
+              </DataList>
+            </span>
+          </div>
+        </div>
+      </ModalBody>
+      <ModalSubmitFooter
+        inProgress={false}
+        cancel={cancel}
+        submitText="Save"
+        resetText="Restore Default Columns"
+        reset={resetColumns}
+      />
+    </form>
+  );
+};
+
+export const createColumnManagementModal = createModalLauncher<ColumnManagementModalProps>(
+  ColumnManagementModal,
+);
+
+ColumnManagementModal.displayName = 'ColumnManagementModal';
+
+type DataListRowProps = {
+  column: any;
+  onChange: (checked: boolean, event: React.SyntheticEvent) => void;
+  disableUncheckedRow: boolean;
+};
+
+export type ColumnManagementModalProps = {
+  cancel?: () => void;
+  close?: () => void;
+  columnManagementID: string;
+  selectedColumns: any;
+  columnManagementType: string;
+};
+
+type ColumnHeaderRow = {
+  ID: string;
```

Comment:
Lowercase `id`

```suggestion
  id: string;
```

Reply:
Removed the need for this type altogether.